src/UpdatedPackagesCollector.js

```diff
@@ -127,7 +127,19 @@ export default class UpdatedPackagesCollector {
 
   collectUpdates() {
     return this.packages.filter(pkg => {
-      return this.updatedPackages[pkg.name] || (this.flags.onlyExplicitUpdates ? false : this.dependents[pkg.name]) || this.flags.canary;
+      if (this.updatedPackages[pkg.name]) {
+        return true;
+      }
+
+      if (!this.flags.onlyExplicitUpdates && this.dependents[pkg.name]) {
+        return true;
+      }
+
+      if (this.flags.canary && !pkg.isPrivate()) {
+        return true;
```

Comment:
I think we should exit early if it's private. The down part with putting the check here is that private packages don't get local dependencies updates when `lerna publish` is called.
e.g

```
packageA
{
  "name": "A",
  "private": true,
  "dependencies": {
    "B": "^1.0.0",
  }
}

packageB
{
  "name": "B"
}
```

if we run `lerna publish` with this patch, the `packageB` dependency in `packageA`  will still be `1.0.0` even if we update `packageB` to `2.0.0`.